src/waitress/channel.py

```diff
@@ -360,88 +375,89 @@ def _flush_outbufs_below_high_watermark(self):
                     self.outbuf_lock.wait()
 
     def service(self):
-        """Execute all pending requests """
-        with self.task_lock:
-            while self.requests:
-                request = self.requests[0]
+        """Execute one request. If there are more, we add another task to the
+        server at the end."""
```

Comment:
This is a noop comment, but just wanted to note that I like this workflow change as it prevents a connection from dominating a thread and tasks will be more fairly queued across connections.